traitsui/key_bindings.py

```diff
@@ -254,12 +254,16 @@ def _focus_owner_changed(self, old, new):
         if old is not None:
             old.border_size = 0
 
-    @on_trait_change("children[]")
-    def _children_modified(self, removed, added):
+    @observe("children.items")
+    def _children_modified(self, event):
         """ Handles child KeyBindings being added to the object.
         """
-        for item in added:
-            item.parent = self
+        if isinstance(event.object, KeyBinding):
+            for item in event.new:
+                item.parent = self
+        else:
+            for item in event.added:
+                item.parent = self
```

Comment:
same comment here - a short comment explaining the difference between the two branches would be useful here i think.